src/cplus_plugin/lib/extent_check.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+"""
+Checks if the current extent is within the pilot area of interest.
+"""
+
+from qgis.core import (
+    QgsCoordinateReferenceSystem,
+    QgsCoordinateTransform,
+    QgsProject,
+    QgsRectangle,
+)
+from qgis.utils import iface
+
+from qgis.PyQt import QtCore, QtGui
+
+from ..conf import settings_manager, Settings
+from ..definitions.defaults import DEFAULT_CRS_ID, PILOT_AREA_EXTENT
+from ..utils import FileUtils, log, tr
+
+
+class PilotExtentCheck(QtCore.QObject):
```

Comment:
We need to add this class functions as utils instead of defining them under the class.

Reply:
Hi @Samweli if we define it as a function then we will not be able to emit any signals when the extent changes thus unable to propagate this to the other widgets listening to this event. But if there are alternatives will be happy to refactor this.

Reply:
Now using a function to check extents.